static/src/javascripts/projects/common/modules/weather.js

```diff
@@ -195,12 +217,32 @@ define([
 
                 // Replace number in weather icon class
                 $weatherIcon.attr('class', $weatherIcon.attr('class').replace(/(\d+)/g,
-                    weatherData.WeatherIcon))
-                    .attr('title', weatherData.WeatherText);
+                    weatherData.weatherIcon))
+                    .attr('title', weatherData.weatherText);
 
                 // Close editing
                 self.toggleControls(false);
             };
+        },
+
+        renderForecast: function (forecastData) {
+            var $forecastHolder = $('.js-weather-forecasts'),
+                $forecast = null,
+                i;
+
+            $forecastHolder.empty();
+
+            for (i in forecastData) {
+                $forecast = $.create(template(forecastTemplate, {
+                    'forecast-time': new Date(forecastData[i].epochDateTime * 1000).getHours(),
+                    'forecast-temp': forecastData[i].temperature[this.getUnits()],
+                    'forecast-icon': forecastData[i].weatherIcon,
+                    'forecast-desc': forecastData[i].weatherText,
+                    'forecast-num': parseInt(i, 10) + 1
+                }));
+
+                $forecast.appendTo($forecastHolder);
```

Comment:
This is going to touch the DOM for each iteration in the loop. Would be better if you create a document fragment `document.createDocumentFragment();` before the loop and append to that.